### DISCUSSION TITLE
Bluetooth/Wifi dongle

### DISCUSSION THREAD

**Discussion by EdNekebno** [EdNekebno/DGTCentaurMods#68] [Ideas]
I thought there was a discussion somewhere that I could tag this on to, but I can't find it. I was tempted to tag it onto the usb gadgets one, but then decided I'd just start a new thread.

At some point we spoke about installs and soldering angled pin headers on the raspberry pi zero w. We've also discussed how you could put a hole in the side of the board for usb gadgets.

When looking there is a shortage of raspberry pi zero (anythings), they're pretty hard to get. I was looking for one. Then I thought - I have a raspberry pi zero - maybe I can get a wifi/bluetooth dongle for it.

Then I was thinking - is that not an alternative option. The zero w is nicer and more reliable for people that feel like soldering. But if we picked a fairly common dongle and made sure it worked with it then people could also choose to put a small hole in the side of the board, change the sd card, and plug in a dongle. Or is that a stupid idea? Something about it seems too simplistic!

There's actually some space in the case between the raspberry pi and the battery. With the right kind of cable and dongle it just might fit...but I think that's probably taking it too far for now. But an external dongle would be a great option.

**Comment by wormstein** [comment]
Sounds like a plan. I think it shouldn't be too hard.

**Comment by EdNekebno** [comment]
I will see what I can find then. I realised what seemed strange about it - normally it would be stupid because the dongle will probably cost more than the pi zero w :) 

Talking of which I got another pi today...I've wasted a whole evening trying to get picochess to work when really I should be doing something more useful :( I can test the build when it's ready.

**Comment by wormstein** [comment]
Not sure what to be ready :) It it's the pi or the build...
But, you can just build one yourself:
```
git clone <repo>
cd DGTCentaurMods/build
./build.sh master
```
You'll find the deb there once it's done.
To do a new build, just `./build.sh clean` and repeat.

**Reply by EdNekebno** [reply]
Yeh...I'll try it. I mean if I get a fresh rasbian lite install and make a deb file then I can install it, switch the file into the DGT centaur and see if things work or work out what commands I need to type to make it work. When we have one that we know works then I think we're ready for that beta release of yours.

**Comment by bpycinski** [comment]
Hello, as there is no way to obtain a fresh rasp zero, probably the best option for using your firmware with a new board would be to attach a wifi dongle. I'd like to know, how much effort does it take to add this functionality to the firmware?

**Comment by wormstein** [comment]
hi @bpycinski , 
attaching an USB dongle for wifi connectivity is possible but you need to make a hole on the side of the board to expose the Raspi's USB port that supports data. That is the left one as you look at the board....

As for coding this in the software... it is possible but not quite easy. Some dongles might require their own kernel modules to be loaded in the OS before the interface is available to the board. Then, we need to figure out a way to identify the wifi interface name as not all dongles will create a wlan0...
If you are a Linux user you might get this setup working for you. Our mod should work on the original Raspi but not be installed using card-setup-tool as that way requires an internet connection during initial setup.

Hope this helps

**Comment by bpycinski** [comment]
Hi @wormstein, thank you for a quick reply. Well, I am a long-time Linux user and an amateur electronic, so I'm fine with your warnings. I have not bought a Centaur yet, and will do it only if I am able to install your software. But do not want to pay additional 100$ for a Pi. 
Forcing the interface name to wlan0 is totally doable, but my main concern is that I would need an additional pi to configure Debian, as I don't want to experiment on the board. But do I understand correctly that if I can set up the wifi interface (i.e. kernel modules, the interface name, the wifi password), I'm free to launch the install script?

**Comment by wormstein** [comment]
Good to hear you are a linux user. You'll need some experience here...
What the install script actually does is to setup and update the Rapberry Pi OS, install our deb file, and move the original centaur software to the pi's home folder. That software is previously copied by the card-setup-tool when the card is prepared.

BUT

The mod itself it's a deb file that once you setup the pi with a new OS, you should be able to just install it - no internet connection is needed this way. It will also confiure the board (enable UART, SPI, etc...)

I think the easiest way to configure a pi without getting another pi with wifi is to enable the serial console and purchase a serial to usb adapter to ssh into the board. I think I saw this tutorial somewhere...
Setup the OS with the dongle first - kernel, interface, wifi - then you can download the deb file and install it.

Cheers

**Comment by bpycinski** [comment]
Hi, thanks a lot for the explanation, it seems much clearer now, and, by the way, looks quite straightforward. Connection with a console through the UART adapter is a good idea. 

When I'll purchase a Centaur and install the mods, will share the experiences.

Greetings!

**Reply by wormstein** [reply]
Join our discord server for support, in case you need it

**Comment by bpycinski** [comment]
In case anybody need it, there is an instruction, how to run the software on original hardware with a wifi+bluetooth dongle: https://github.com/bpycinski/DGTCentaurMods/wiki/Deploy-the-Mods-on-the-original-hardware
